drivers/storage/portworx/components_test.go

```diff
@@ -2794,6 +2803,128 @@ func TestSecurityInstall(t *testing.T) {
+	mockClusterServer := mock.NewMockOpenStorageClusterServer(mockCtrl)
+	mockNodeServer := mock.NewMockOpenStorageNodeServer(mockCtrl)
+
+	// Start a sdk server that implements the mock servers
+	sdkServerIP := "127.0.0.1"
+	sdkServerPort := 23888
+	mockSdk := mock.NewSdkServer(mock.SdkServers{
+		Role:    mockRoleServer,
+		Node:    mockNodeServer,
+		Cluster: mockClusterServer,
+	})
+	err := mockSdk.StartOnAddress(sdkServerIP, strconv.Itoa(sdkServerPort))
+	require.NoError(t, err)
+	defer mockSdk.Stop()
+
+	k8sClient := testutil.FakeK8sClient(&v1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      pxutil.PortworxServiceName,
+			Namespace: "kube-test",
+		},
+		Spec: v1.ServiceSpec{
+			ClusterIP: sdkServerIP,
+			Ports: []v1.ServicePort{
+				{
+					Name: pxutil.PortworxSDKPortName,
+					Port: int32(sdkServerPort),
+				},
+			},
+		},
+	})
+	driver := portworx{}
+	driver.Init(k8sClient, runtime.NewScheme(), record.NewFakeRecorder(10))
+	cluster := &corev1alpha1.StorageCluster{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "px-cluster",
+			Namespace: "kube-test",
+		},
+		Spec: corev1alpha1.StorageClusterSpec{
+			Security: &corev1alpha1.SecuritySpec{
+				Enabled: true,
+				Auth: &corev1alpha1.AuthSpec{
+					GuestAccess: &corev1alpha1.GuestRoleManaged,
+					SelfSigned: &corev1alpha1.SelfSignedSpec{
+						TokenLifetime: &metav1.Duration{
+							// Since we have a token expiration buffer of one minute,
+							// a new token will constantly be fetched.
+							Duration: time.Second * 1,
+						},
+					},
+				},
+			},
+		},
+	}
+	// Initial run
+	setSecuritySpecDefaults(cluster)
+	securityComponent, ok := component.Get(component.SecurityComponentName)
+	require.True(t, ok)
+	component.DeregisterAllComponents()
+	component.Register(component.SecurityComponentName, securityComponent)
+	securityComponent.Initialize(k8sClient, hversion.Version{}, runtime.NewScheme(), record.NewFakeRecorder(10))
+	driver.UpdateStorageClusterStatus(cluster)
+	require.NoError(t, err)
+
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	// Disable GuestAccess. Should expect an update to be called.
+	cluster.Spec.Security.Auth.GuestAccess = &corev1alpha1.GuestRoleDisabled
+	expectedRole := component.GuestRoleEnabled
+	expectedRoleInspectResp := &osdapi.SdkRoleInspectResponse{
+		Role: &expectedRole,
+	}
+	mockRoleServer.EXPECT().
+		Inspect(gomock.Any(), &osdapi.SdkRoleInspectRequest{
+			Name: component.SecuritySystemGuestRoleName,
+		}).
+		Return(expectedRoleInspectResp, nil).
+		Times(1)
+	mockRoleServer.EXPECT().
+		Update(gomock.Any(), gomock.Any()).
+		Return(&osdapi.SdkRoleUpdateResponse{
+			Role: &component.GuestRoleDisabled,
+		}, nil).
+		Times(1)
+	err = driver.PreInstall(cluster)
+	require.NoError(t, err)
+
+	// Enable guest access, should be updated again
+	cluster.Spec.Security.Auth.GuestAccess = &corev1alpha1.GuestRoleEnabled
+	expectedRole = component.GuestRoleEnabled
```

Comment:
Shouldn't this be GuestRoleDisabled? 
Also how is the test passing of both expected and desired are the same?

Reply:
In this section, I'm setting GuestRoleEnabled in the cluster spec, and I'm expecting that during a PreInstall, we will make a call to the Mock server with the GuestRoleEnabled RBAC.